apps/ledger-live-mobile/src/newArch/features/LandingPages/screens/GenericLandingPage/index.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect } from "react";
+import { Flex } from "@ledgerhq/native-ui";
+import { LandingPagesNavigatorParamList } from "~/components/RootNavigator/types/LandingPagesNavigator";
+import { BaseComposite, StackNavigatorProps } from "~/components/RootNavigator/types/helpers";
+import { ScreenName } from "~/const";
+import ContentCardsLocation from "~/dynamicContent/ContentCardsLocation";
+import { TrackScreen } from "~/analytics";
+import { LandingPageUseCase } from "~/dynamicContent/types";
+import useDynamicContent from "~/dynamicContent/useDynamicContent";
+import { filterCategoriesByLocation } from "~/dynamicContent/utils";
+
+type NavigationProps = BaseComposite<
+  StackNavigatorProps<LandingPagesNavigatorParamList, ScreenName.GenericLandingPage>
+>;
+
+const GenericLandingPage = (props: NavigationProps) => {
+  const useCase = props.route.params?.useCase;
+  const { categoriesCards } = useDynamicContent();
+
+  useEffect(() => {
+    if (!useCase || !Object.values(LandingPageUseCase).includes(useCase)) {
+      props.navigation.goBack();
+    }
+    const categoriesToDisplay = filterCategoriesByLocation(categoriesCards, useCase);
+    if (categoriesToDisplay.length === 0) {
+      props.navigation.goBack();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We should not lie to React, most of the time when we need to, it's a conceptual issue. Can you rework that part, please?
https://react.dev/learn/removing-effect-dependencies